Employee/EmployeeManagement.java

```diff
@@ -0,0 +1,366 @@
+                    employee.setId(Integer.parseInt(parts[0]));
+                    employee.setName(parts[1]);
+
+                    
+                    employee.setEmail(parts[2]);    
+                    
+                    employee.setAge(Integer.parseInt(parts[3]));
+                    try {
+                        employee.setDob(dateFormat.parse(parts[4]));
+                    } catch (ParseException e) {
+                        e.printStackTrace();
+                        System.out.println( "Invalid Date");
+                    }
+                    employees.add(employee);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            // If the file is not found, just return an empty list
+        }
+        return employees;
+    }
+
+    private static void saveEmployeesToFile(List<Employee> employees) {
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+
+        try (PrintWriter writer = new PrintWriter(new FileWriter(FILE_NAME))) {
+            for (Employee employee : employees) {
+                writer.println(employee.getId() + "," + employee.getName() + "," + employee.getEmail() + "," + employee.getAge() + "," + dateFormat.format(employee.getDob()));
+            }
+        } catch (IOException e) {
+            System.out.println("Error saving employees to file");
+        }
+    }
+
+    private static void addEmployee(Scanner scanner) {
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+        dateFormat.setLenient(false);
+
+        Employee employee = new Employee();
+        System.out.print("Enter name: ");
+        String name = scanner.nextLine();
+        if (name.matches("[a-zA-Z ]+") && !name.isEmpty()) {
+            employee.setName(name);
+        } else {
+            System.out.println("Invalid name");
+            return;
+        }
+    
+
+
+        System.out.print("Enter email address: ");
+        String email = scanner.nextLine();
+        if(isValidEmail(email))
+        employee.setEmail(email);
+        else{
+            System.out.println("Invalid Email");
+            System.exit(0);
+        }
+
+
+        System.out.print("Enter age: ");
+        int age = scanner.nextInt();
+        scanner.nextLine();
+        if (age >= 18 && age <= 60) {
+            employee.setAge(age);
+        } else {
+            System.out.println("Invalid age");
+            return;
+        }
+
+        System.out.print("Enter date of birth (dd-mm-yyyy): ");
+        try {
+            employee.setDob(dateFormat.parse(scanner.nextLine()));
+        } catch (ParseException e) {
+            System.out.println("Invalid date format");
+            e.printStackTrace();
+        }
+        employee.setId(generateEmployeeId());
+        employees.add(employee);
+        saveEmployeesToFile(employees);
+
+        System.out.println("Employee added successfully");
+    }
+
+    private static int generateEmployeeId() {
```

Comment:
Can we maintain a static variable for id, because this method will run with O(n) complexity every time we add a new employee?